besu/src/main/java/org/hyperledger/besu/services/TraceServiceImpl.java

```diff
@@ -85,10 +87,41 @@ public void traceBlock(final long blockNumber, final BlockAwareOperationTracer t
    * @param tracer an instance of OperationTracer
    */
   @Override
-  public void traceBlock(final Hash hash, final BlockAwareOperationTracer tracer) {
+  public BlockTraceResult traceBlock(final Hash hash, final BlockAwareOperationTracer tracer) {
+    return traceBlock(blockchainQueries.getBlockchain().getBlockByHash(hash), tracer);
+  }
+
+  private BlockTraceResult traceBlock(
+      final Optional<Block> maybeBlock, final BlockAwareOperationTracer tracer) {
     checkArgument(tracer != null);
-    final Optional<Block> block = blockchainQueries.getBlockchain().getBlockByHash(hash);
-    block.ifPresent(value -> trace(value, tracer));
+    if (maybeBlock.isEmpty()) {
+      return BlockTraceResult.empty();
+    }
+
+    final Optional<List<TransactionProcessingResult>> results = trace(maybeBlock.get(), tracer);
```

Comment:
is it more efficient to assign a variable to maybeBlock.get() - it's used below as well